turbopack/crates/turbo-tasks-backend/src/backend/mod.rs

```diff
@@ -852,26 +852,31 @@ impl<B: BackingStorage> TurboTasksBackendInner<B> {
                 return (None, None);
             }
             let len = inner.len();
-            let mut meta = Vec::with_capacity(len);
-            let mut data = Vec::with_capacity(len);
+
+            let meta_restored = inner.state().meta_restored();
+            let data_restored = inner.state().data_restored();
+
+            let mut meta = Vec::with_capacity(if meta_restored { len } else { 0 });
```

Comment:
nit: use the same pattern as line 893 below?

i suppose `len` is an over-estimate by at least half for one of the two lists?  probab